GoogleForCreators/web-stories-wp
# Dashboard: Allow users to filter/search by author in the story dashboard

## Context
An old feature request finally coming to fruition


## Summary
Adds the same author dropdown as in the editor to filter stories with in the dashboard.


## Relevant Technical Choices
Had to pull the `AdvancedDropDown` from the editor into the design system to use it here since this is the only implementation of this datalist like component. The unit tests were tightly coupled with the font picker implementation, so I just moved the unit tests to there.


## To-do



## User-facing changes
Now there's an author filter in the dashboard:

<img width="1283" alt="Screen Shot 2021-10-26 at 3 40 00 PM" src="https://user-images.githubusercontent.com/35983235/138957539-e0094ab6-3a42-4960-b3bf-01f8da8296f4.png">



## Testing Instructions






This PR can be tested by following these steps:

1. Open up the dashboard
2. See that you can filter stories by selecting different authors in this dropdown:
<img width="1283" alt="Screen Shot 2021-10-26 at 3 40 00 PM" src="https://user-images.githubusercontent.com/35983235/138957629-d4fbeb89-d780-451f-affe-bcdbd6476189.png">
3. See that you can remove the author filter by clicking on the selected author

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #1872
